### ISSUE TITLE
Use sklearn's Gradient Boosting instead of XGBoost

### ISSUE THREAD

**Issue by ClimbsRocks** [ClimbsRocks/auto_ml#55]
As noted in https://github.com/dmlc/xgboost/issues/1091 XGBoost's support for sparse matrices is a bit... sparse :)

Until that's addressed, use sklearn's gradient boosting instead. It's not as good, but it'll have to do for now.

**Comment by ClimbsRocks** [comment]
Done. sklearn's version seems to be working reasonably well, though they don't support sparse matrices for prediction yet. there's a fix in the works though, so we'll stick with this for now since it's a relatively easy workaround.
